packages/flutter/test/material/text_field_test.dart

```diff
@@ -92,19 +92,21 @@ Widget overlayWithEntry(OverlayEntry entry) {
 }
 
 Widget boilerplate({ Widget child }) {
-  return Localizations(
-    locale: const Locale('en', 'US'),
-    delegates: <LocalizationsDelegate<dynamic>>[
-      WidgetsLocalizationsDelegate(),
-      MaterialLocalizationsDelegate(),
-    ],
-    child: Directionality(
-      textDirection: TextDirection.ltr,
-      child: MediaQuery(
-        data: const MediaQueryData(size: Size(800.0, 600.0)),
-        child: Center(
-          child: Material(
-            child: child,
+  return MaterialApp(
```

Comment:
I think it is good to wrap textfield in overlay, but I am curious why we need to do it here and all the other test.

Reply:
These tests started failing after I started calling `_handleSelectionChanged`.  The reason is that now the selection menu is being created, and it asserts that an Overlay exists above it.  Without the MaterialApp there was no Overlay, and the assertion would trigger.